classes/add_course_form.php

```diff
@@ -28,21 +28,44 @@
 class add_course_form extends moodleform {
 
     protected function definition () {
+    }
+
+    public function init ($id) {
         // Variable violates moodle codestyle but this is required by the lsf-plugin.
-        global $pgDB, $USER; // phpcs:ignore // @codingStandardsIgnoreLine
+        global $pgDB, $USER, $COURSE; // phpcs:ignore // @codingStandardsIgnoreLine
         $mform = $this->_form;
 
         $mform->addElement('html', '<h3>'. get_string('add_course_header', 'block_evasys_sync') .'</h3>');
         $pgDB = new \pg_lite(); // phpcs:ignore // @codingStandardsIgnoreLine
         $pgDB->connect(); // phpcs:ignore // @codingStandardsIgnoreLine
-        $availablecourselist = get_teachers_course_list($USER->username, false);
+        global $DB;
+        $lsfid = $DB->get_field('course', 'idnumber', array('id' => $id));
+        $maincourse = (array) get_course_by_veranstid($lsfid);
+        $veranstids = get_veranstids_by_teacher(get_teachers_pid($USER->username));
+        $courses = get_courses_by_veranstids($veranstids);
+        $availablecourselist = array();
+        foreach ($courses as $veranstid => $course) {
+            $result = array();
+            $result['veranstid'] = $course->veranstid;
+            $result['info'] = $course->titel;
+            $result['semestertxt'] = $course->semestertxt;
+            $result['semester'] = $course->semester;
+            $availablecourselist[$course->veranstid] = $result;
+            $sorter = function ($a, $b) {
+                return $a['semester'] < $b['semester'];
```

Comment:
The sort function for usort should return a positive value if a > b, 0 if a == b and a negative value if a < b.
I don't know what exactly $course->semester is, if it is a number, you can just use $b['semester'] - $a['semester'] (that way around it sorts descendingly).

It seems like a boolean comparison does also work,(so just returns 1 for a>b and 0 otherwise) but if it's specified that way, I think we should do it. https://www.php.net/manual/de/function.usort.php

Reply:
The manual also specifies that boolean values will be cast to Integer, thats why I chose this way since I belive It's more readable.

Reply:
Yes, but in your case with sorting descendingly, a < b returns 1 if a < b, 0 if a = b and 0 if a > b. The function however should return a negative value if a > b.